### PR TITLE
ADD: authentication and authorization user doc

### PR DESCRIPTION
**Pull request by xuanyu66**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tispark/pull/2185

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by birdstorm — APPROVED** [review]
LGTM

**Review by sunxiaoguang — APPROVED** [review]
LGTM

File: tispark-overview.md [hunk 1]
```diff
@@ -355,6 +355,25 @@ TiSpark 可以使用 TiDB 的统计信息：
 | --------   | -----:   | :----: |
 | spark.tispark.statistics.auto_load | true | 是否默认进行统计信息读取 |
 
+## 安全
+
+目前 TiSpark (version >= 2.5.0) 可以通过 TiDB 进行认证与鉴权，该功能默认关闭
```
Comment (qiancai) [inline tispark-overview.md:360]:
```suggestion
从 TiSpark v2.5.0 起，你可以通过 TiDB 对 TiSpark 进行认证与鉴权。该功能默认关闭。
```

File: tispark-overview.md [hunk 2]
```diff
@@ -355,6 +355,25 @@ TiSpark 可以使用 TiDB 的统计信息：
...
+
+在 Spark 配置文件 `spark-defaults.conf` 中添加以下配置项：
```
Comment (qiancai) [inline tispark-overview.md:362]:
```suggestion
要开启认证与鉴权功能，请在 Spark 配置文件 `spark-defaults.conf` 中添加以下配置项：
```

File: tispark-overview.md [hunk 3]
```diff
@@ -355,6 +355,25 @@ TiSpark 可以使用 TiDB 的统计信息：
...
+spark.sql.auth.enable   true
+
+// 配置 TiDB 信息 
+spark.sql.tidb.addr    $your_tidb_server_address
+spark.sql.tidb.port    $your_tidb_server_port
+spark.sql.tidb.user    $your_tidb_server_user
+spark.sql.tidb.password $your_tidb_server_password
+```
+
+详细使用手册请阅读[这份文档](https://github.com/pingcap/tispark/blob/master/docs/authorization_userguide.md)。
```
Comment (qiancai) [inline tispark-overview.md:375]:
```suggestion
详细使用手册，请阅读[这份文档](https://github.com/pingcap/tispark/blob/master/docs/authorization_userguide.md)。
```

**Comment by qiancai** [comment]
@xuanyu66  目前 TiSpark (version >= 2.5.0) 可以通过 TiDB 进行认证与鉴权”，请问 TiSpark 2.5.0 是用来支持 Spark 2.5.x 的吗？ 如果是，这个环境准备章节需要增加下版本信息

https://docs.pingcap.com/zh/tidb/dev/tispark-overview#%E7%8E%AF%E5%A2%83%E5%87%86%E5%A4%87

**Comment by zhangyangyu** [comment]
> @xuanyu66 目前 TiSpark (version >= 2.5.0) 可以通过 TiDB 进行认证与鉴权”，请问 TiSpark 2.5.0 是用来支持 Spark 2.5.x 的吗？ 如果是，这个环境准备章节需要增加下版本信息
> 
> https://docs.pingcap.com/zh/tidb/dev/tispark-overview#%E7%8E%AF%E5%A2%83%E5%87%86%E5%A4%87

直接把 https://github.com/pingcap/tispark#how-to-choose-tispark-version 这个表格放过来吧，清晰明了一些。

**Review by qiancai — APPROVED** [review]

File: tispark-overview.md [hunk 4]
```diff
@@ -355,6 +355,25 @@ TiSpark 可以使用 TiDB 的统计信息：
...
 | spark.tispark.statistics.auto_load | true | 是否默认进行统计信息读取 |
 
+## 安全
+
+从 TiSpark v2.5.0 起，你可以通过 TiDB 对 TiSpark 进行认证与鉴权。该功能默认关闭。
+
+要开启认证与鉴权功能，请在 Spark 配置文件 `spark-defaults.conf` 中添加以下配置项：
+
+```
+// 打开认证授权功能
```
Comment (qiancai) [inline tispark-overview.md:365]:
```suggestion
// 开启认证授权功能
```

File: tispark-overview.md [hunk 5]
```diff
@@ -23,9 +23,12 @@ TiSpark 是将 Spark SQL 直接运行在分布式存储引擎 TiKV 上的 OLAP 
 
 ## 环境准备
 
-现有 TiSpark 2.x 版本支持 Spark 2.3.x 和 Spark 2.4.x。如果你希望使用 Spark 2.1.x 版本，需使用 TiSpark 1.x。
+现有 TiSpark 版本兼容情况如下图所示，用户可以根据自身情况选择相应 TiSpark 版本。
```
Comment (xuanyu66) [inline tispark-overview.md:26]:
```suggestion
目前 TiSpark 支持版本的兼容情况如下图所示，用户可以根据自身情况选择相应 TiSpark 版本。
```

File: tispark-overview.md [hunk 6]
```diff
@@ -23,9 +23,12 @@ TiSpark 是将 Spark SQL 直接运行在分布式存储引擎 TiKV 上的 OLAP 
 
 ## 环境准备
 
-现有 TiSpark 2.x 版本支持 Spark 2.3.x 和 Spark 2.4.x。如果你希望使用 Spark 2.1.x 版本，需使用 TiSpark 1.x。
+目前 TiSpark 支持版本的兼容情况如下图所示，用户可以根据自身情况选择相应 TiSpark 版本。
 
-TiSpark 需要 JDK 1.8+ 以及 Scala 2.11（Spark2.0+ 默认 Scala 版本）。
+| TiSpark Version | TiDB/TiKV/PD Version | Spark Version | Scala Version |
```
Comment (qiancai) [inline tispark-overview.md:28]:
```suggestion
| TiSpark 版本 | TiDB、TiKV、PD 版本 | Spark 版本 | Scala 版本 |
```

File: tispark-overview.md [hunk 7]
```diff
@@ -23,9 +23,12 @@ TiSpark 是将 Spark SQL 直接运行在分布式存储引擎 TiKV 上的 OLAP 
 
 ## 环境准备
 
-现有 TiSpark 2.x 版本支持 Spark 2.3.x 和 Spark 2.4.x。如果你希望使用 Spark 2.1.x 版本，需使用 TiSpark 1.x。
+目前 TiSpark 支持版本的兼容情况如下图所示，用户可以根据自身情况选择相应 TiSpark 版本。
```
Comment (qiancai) [inline tispark-overview.md:26]:
```suggestion
目前 TiSpark 支持版本的兼容情况如下所示，你可以根据自身情况选择相应的 TiSpark 版本。
```

File: tispark-overview.md [hunk 8]
```diff
@@ -355,6 +358,25 @@ TiSpark 可以使用 TiDB 的统计信息：
 | --------   | -----:   | :----: |
 | spark.tispark.statistics.auto_load | true | 是否默认进行统计信息读取 |
 
+## 安全
+
+从 TiSpark v2.5.0 起，你可以通过 TiDB 对 TiSpark 进行认证与鉴权。该功能默认关闭。
```
Comment (qiancai) [inline tispark-overview.md:363]:
```suggestion
从 TiSpark v2.5.0 起，你可以通过 TiDB 对 TiSpark 进行鉴权与授权。该功能默认关闭。
```

File: tispark-overview.md [hunk 9]
```diff
@@ -355,6 +358,25 @@ TiSpark 可以使用 TiDB 的统计信息：
...
+spark.sql.auth.enable   true
+
+// 配置 TiDB 信息 
+spark.sql.tidb.addr    $your_tidb_server_address
+spark.sql.tidb.port    $your_tidb_server_port
+spark.sql.tidb.user    $your_tidb_server_user
+spark.sql.tidb.password $your_tidb_server_password
+```
+
+详细使用手册，请参考[这份文档](https://github.com/pingcap/tispark/blob/master/docs/authorization_userguide.md)。
```
Comment (qiancai) [inline tispark-overview.md:378]:
```suggestion
更多详细信息，请参考[TiSpark 鉴权与授权文档](https://github.com/pingcap/tispark/blob/master/docs/authorization_userguide.md)。
```

File: tispark-overview.md [hunk 10]
```diff
@@ -355,6 +358,25 @@ TiSpark 可以使用 TiDB 的统计信息：
 | --------   | -----:   | :----: |
 | spark.tispark.statistics.auto_load | true | 是否默认进行统计信息读取 |
 
+## 安全
+
+从 TiSpark v2.5.0 起，你可以通过 TiDB 对 TiSpark 进行认证与鉴权。该功能默认关闭。
+
+要开启认证与鉴权功能，请在 Spark 配置文件 `spark-defaults.conf` 中添加以下配置项：
```
Comment (qiancai) [inline tispark-overview.md:365]:
```suggestion
要开启鉴权与授权功能，请在 Spark 配置文件 `spark-defaults.conf` 中添加以下配置项：
```

File: tispark-overview.md [hunk 11]
```diff
@@ -355,6 +358,25 @@ TiSpark 可以使用 TiDB 的统计信息：
...
+
+```
+// 开启认证授权功能
```
Comment (qiancai) [inline tispark-overview.md:368]:
```suggestion
// 开启鉴权与授权功能
```

File: tispark-overview.md [hunk 12]
```diff
@@ -23,9 +23,12 @@ TiSpark 是将 Spark SQL 直接运行在分布式存储引擎 TiKV 上的 OLAP 
...
 ## 环境准备
 
-现有 TiSpark 2.x 版本支持 Spark 2.3.x 和 Spark 2.4.x。如果你希望使用 Spark 2.1.x 版本，需使用 TiSpark 1.x。
+目前 TiSpark 支持版本的兼容情况如下所示，你可以根据自身情况选择相应的 TiSpark 版本。
 
-TiSpark 需要 JDK 1.8+ 以及 Scala 2.11（Spark2.0+ 默认 Scala 版本）。
+| TiSpark 版本 | TiDB、TiKV、PD 版本 | Spark 版本 | Scala 版本 |
+| --------------- | -------------------- | ------------- | ------------- |
+| 2.4.x           | 5.x, 4.x             | 2.3.x, 2.4.x  | 2.11          |
+| 2.5.x           | 5.x, 4.x             | 3.0.x, 3.1.x  | 2.12          |
```
Comment (qiancai) [inline tispark-overview.md:31]:
```suggestion
| 2.4.x           | 5.x，4.x             | 2.3.x，2.4.x  | 2.11          |
| 2.5.x           | 5.x，4.x             | 3.0.x，3.1.x  | 2.12          |
```

File: tispark-overview.md [hunk 13]
```diff
@@ -355,6 +358,25 @@ TiSpark 可以使用 TiDB 的统计信息：
...
+spark.sql.auth.enable   true
+
+// 配置 TiDB 信息 
+spark.sql.tidb.addr    $your_tidb_server_address
+spark.sql.tidb.port    $your_tidb_server_port
+spark.sql.tidb.user    $your_tidb_server_user
+spark.sql.tidb.password $your_tidb_server_password
+```
+
+更多详细信息，请参考[TiSpark 鉴权与授权文档](https://github.com/pingcap/tispark/blob/master/docs/authorization_userguide.md)。
```
Comment (qiancai) [inline tispark-overview.md:378]:
```suggestion
更多详细信息，请参考 [TiSpark 鉴权与授权文档](https://github.com/pingcap/tispark/blob/master/docs/authorization_userguide.md)。
```

File: tispark-overview.md [hunk 14]
```diff
@@ -23,9 +23,12 @@ TiSpark 是将 Spark SQL 直接运行在分布式存储引擎 TiKV 上的 OLAP 
 
 ## 环境准备
 
-现有 TiSpark 2.x 版本支持 Spark 2.3.x 和 Spark 2.4.x。如果你希望使用 Spark 2.1.x 版本，需使用 TiSpark 1.x。
+目前 TiSpark 支持版本的兼容情况如下所示，你可以根据自身情况选择相应的 TiSpark 版本。
```
Comment (shichun-0415) [inline tispark-overview.md:26]:
```suggestion
目前 TiSpark 支持版本的兼容情况如下所示，你可以根据实际情况选择相应的 TiSpark 版本。
```

File: tispark-overview.md [hunk 15]
```diff
@@ -355,6 +358,25 @@ TiSpark 可以使用 TiDB 的统计信息：
 | --------   | -----:   | :----: |
 | spark.tispark.statistics.auto_load | true | 是否默认进行统计信息读取 |
 
+## 安全
+
+从 TiSpark v2.5.0 起，你可以通过 TiDB 对 TiSpark 进行鉴权与授权。该功能默认关闭。
+
+要开启鉴权与授权功能，请在 Spark 配置文件 `spark-defaults.conf` 中添加以下配置项：
```
Comment (shichun-0415) [inline tispark-overview.md:365]:
```suggestion
从 TiSpark v2.5.0 起，你可以通过 TiDB 对 TiSpark 进行鉴权与授权。

该功能默认关闭，要开启该功能，请在 Spark 配置文件 `spark-defaults.conf` 中添加以下配置项：
```

**Review by shichun-0415 — APPROVED** [review]
rest LGTM

File: tispark-overview.md [hunk 16]
```diff
@@ -23,9 +23,12 @@ TiSpark 是将 Spark SQL 直接运行在分布式存储引擎 TiKV 上的 OLAP 
 
 ## 环境准备
 
-现有 TiSpark 2.x 版本支持 Spark 2.3.x 和 Spark 2.4.x。如果你希望使用 Spark 2.1.x 版本，需使用 TiSpark 1.x。
+目前 TiSpark 支持版本的兼容情况如下所示，你可以根据实际情况选择相应的 TiSpark 版本。
```
Comment (qiancai) [inline tispark-overview.md:26]:
```suggestion
目前 TiSpark 支持版本的兼容情况如下所示，你可以根据需要选择相应的 TiSpark 版本。
```

File: tispark-overview.md [hunk 17]
```diff
@@ -355,6 +358,25 @@ TiSpark 可以使用 TiDB 的统计信息：
 | --------   | -----:   | :----: |
 | spark.tispark.statistics.auto_load | true | 是否默认进行统计信息读取 |
 
+## 安全
+
+从 TiSpark v2.5.0 起，你可以通过 TiDB 对 TiSpark 进行鉴权与授权。
+
+该功能默认关闭，要开启该功能，请在 Spark 配置文件 `spark-defaults.conf` 中添加以下配置项：
```
Comment (qiancai) [inline tispark-overview.md:365]:
```suggestion
该功能默认关闭。要开启该功能，请在 Spark 配置文件 `spark-defaults.conf` 中添加以下配置项：
```

**Pull request merged**